terraform/templates/windows_worker_user_data.tpl

```diff
@@ -42,11 +42,19 @@
   # Add config to HabService.dll.config
   $svcPath = Join-Path $env:SystemDrive "hab\svc\windows-service"
   [xml]$configXml = Get-Content (Join-Path $svcPath HabService.dll.config)
-  $configXml.configuration.appSettings.add[1].value = "${flags}"
-  $configXml.Save((Join-Path $svcPath HabService.dll.config))
 
-  # Enable PIDS_FROM_LAUNCHER feature
-  $env:HAB_FEAT_PIDS_FROM_LAUNCHER=1;
+  # Update the arguments for the Supervisor
+  $launcherArgs = $configxml | select-xml -xpath "//appSettings/add[@key='launcherArgs']"
+  $launcherArgs.Node.value = "${flags}"
+
+%{ for feature in enabled_features ~}
+  $child = $configXml.CreateElement("add")
+  $child.SetAttribute("key", "ENV_HAB_FEAT_${upper(feature)}")
+  $child.SetAttribute("value", "true")
+  $configXml.configuration.appSettings.AppendChild($child)
+
```

Comment:
Remove blank line to be consistent

Reply:
I left the blank lines like this because it results in the rendered file being nicely delimited, which makes it easier to validate by eye.

If there are no enabled features, the resulting text looks like this:
```powershell
  ...
  # Update the arguments for the Supervisor
  $launcherArgs = $configxml | select-xml -xpath "//appSettings/add[@key='launcherArgs']"
  $launcherArgs.Node.value = "${flags}"

  $configXml.Save((Join-Path $svcPath HabService.dll.config))
  ...
```
If there are enabled features, they each get their own "stanza":
```powershell
  ...
  # Update the arguments for the Supervisor
  $launcherArgs = $configxml | select-xml -xpath "//appSettings/add[@key='launcherArgs']"
  $launcherArgs.Node.value = "${flags}"

  $child = $configXml.CreateElement("add")
  $child.SetAttribute("key", "ENV_HAB_FEAT_FOO")
  $child.SetAttribute("value", "true")
  $configXml.configuration.appSettings.AppendChild($child)

  $child = $configXml.CreateElement("add")
  $child.SetAttribute("key", "ENV_HAB_FEAT_BAR")
  $child.SetAttribute("value", "true")
  $configXml.configuration.appSettings.AppendChild($child)

  $configXml.Save((Join-Path $svcPath HabService.dll.config))
  ...
```
It certainly doesn't make a difference when the file is being consumed, but it came in handy when I was testing this out using `terraform console`. 

That being said, I'm happy to change it to whatever you prefer.

Reply:
:+1:  Thanks for the detailed explanation, makes sense!